lib/decoder.ts

```diff
@@ -108,7 +108,7 @@ class Decoder {
     }
     this.verifyClaims();
 
-    return this._body;
+    return this._body as JWTBody<T>;
```

Comment:
Using `as` in TypeScript is generally an anti-pattern. You are needing to do this as the Decoder class is no longer correctly typed.

It would be better to pass the `T` generic to the class itself, then you don't have to use this hack and can apply it to the class variable definition instead.

```ts
class Decoder<T> {
  _header: JWTHeader;
  _body: JWTBody<T>;
  
  ...
  
   decodeAndVerify(token: JWTToken, options: DecodingOptions = {}): JWTBody<T> {
     ...
     return this._body;
  }
```

Then in the index.ts you can simply pass the type as:

```ts
const decoder = new Decoder<T>(key);
return decoder.decodeAndVerify(token, options);
```